packages/logger/src/logger.test.ts

```diff
@@ -0,0 +1,93 @@
+import { describe, expect, it, vi } from "vitest";
+import { createLogger } from "./logger";
+import { attachLoggerConsoleTransport } from "./logger-console-transport";
+import { attachLoggerOtelTransport } from "./logger-otel-transport";
+import { logs } from "@opentelemetry/api-logs";
+import { SemanticResourceAttributes } from "@opentelemetry/semantic-conventions";
+
+vi.spyOn(console, "log");
+vi.setSystemTime(new Date(2024, 1, 1, 5, 15));
+
+describe("Logger", () => {
+  describe("Console Transport", () => {
+    it("Prints message and nested object to the console, including attributes passed from the parent scope", () => {
+      const logger = createLogger("Test Logger", {
+        rootScopePrimitiveArg: 1,
+        rootScopeObjectArg: {
+          objectKey: "objectValue",
+        },
+      });
+
+      attachLoggerConsoleTransport(logger);
+
+      logger.info("Test Message", {
+        childScopePrimitiveArg: 2,
+        childScopeObjectArg: {
+          objectKey: "objectValue",
+        },
+      });
+
+      /**
+       * TODO this assertion fails, but strings are the same. Why?
+       */
+      expect(console.log).toBeCalledWith(" 2024-02-01T04:15:00.000Z :Test Logger \tTest Message", {
```

Comment:
Can you try to run it locally? This assertion fails, but strings are equal.

Reply:
I've played a bit with this an apparently this string will work:

```
[2m 2024-02-01T04:15:00.000Z :Test Logger[0m 	Test Message
```

I think we can also use [toMatchSnapshot()](https://vitest.dev/api/expect#tomatchsnapshot) instead 🤔

Reply:
`"\x1B[2m 2024-02-01T04:15:00.000Z :Test Logger\x1B[0m \tTest Message"` worked for me

Reply:
Thanks, snapshot works too. I updated this, forgot about color characters